fdbcli/fdbcli.actor.cpp

```diff
@@ -751,9 +767,9 @@ void printStatus(StatusObjectReader statusObj, StatusClient::StatusLevel level,
 							fatalRecoveryState = true;
 
 							if (name == "recruiting_transaction_servers") {
-								description += format("\nNeed at least %d log servers, %d proxies and %d resolvers.", recoveryState["required_logs"].get_int(), recoveryState["required_proxies"].get_int(), recoveryState["required_resolvers"].get_int());
+								description += format("\nNeed at least %d log servers across unique zones, %d proxies and %d resolvers.", recoveryState["required_logs"].get_int(), recoveryState["required_proxies"].get_int(), recoveryState["required_resolvers"].get_int());
```

Comment:
What are the rules for proxies and resolvers? Can they be recruited in the same zone? On the same machine?

Reply:
Proxy and Resolver does not have this constraint.

Reply:
Do they have any constraints?

Reply:
They should be on non-excluded workers. That is the only constraint.

Reply:
And presumably on separate processes (i.e. there won't be two proxies on the same process). They can be on the same machine though?

Reply:
Yes. They can be on same machine.